test/model/autocomplete_test.dart

```diff
@@ -752,43 +756,129 @@ void main() {
           .throws<AssertionError>();
       });
     });
 
+    group('computeWildcardMentionResults', () {
+      Iterable<WildcardMentionOption> getWildcardOptionsFor(String rawQuery, {
+        bool isSilent = false,
+        required Narrow narrow,
+        int? zulipFeatureLevel,
+      }) {
+        final store = eg.store(
+          account: eg.account(user: eg.selfUser, zulipFeatureLevel: zulipFeatureLevel),
+          initialSnapshot: eg.initialSnapshot(zulipFeatureLevel: zulipFeatureLevel));
+        final view = MentionAutocompleteView.init(store: store, localizations: localizations,
+          narrow: narrow, query: MentionAutocompleteQuery(rawQuery, silent: isSilent));
+        final results = <MentionAutocompleteResult>[];
+        view.computeWildcardMentionResults(results: results,
+          isComposingChannelMessage: narrow is ChannelNarrow
+            || narrow is TopicNarrow);
+        view.dispose();
+        return results.map((e) => (e as WildcardMentionAutocompleteResult).wildcardOption);
+      }
+
+      const channelNarrow = ChannelNarrow(1);
+      const topicNarrow = TopicNarrow(1, TopicName('topic'));
+      final dmNarrow = DmNarrow.withUser(10, selfUserId: 5);
+
+      final testCases = [
+        ('',          channelNarrow, [WildcardMentionOption.all, WildcardMentionOption.topic]),
+        ('',          topicNarrow,   [WildcardMentionOption.all, WildcardMentionOption.topic]),
+        ('',          dmNarrow,      [WildcardMentionOption.all]),
+
+        ('c',         channelNarrow, [WildcardMentionOption.channel, WildcardMentionOption.topic]),
+        ('ch',        topicNarrow,   [WildcardMentionOption.channel]),
+        ('str',       channelNarrow, [WildcardMentionOption.stream]),
+        ('e',         topicNarrow,   [WildcardMentionOption.everyone]),
+        ('everyone',  channelNarrow, [WildcardMentionOption.everyone]),
+        ('t',         topicNarrow,   [WildcardMentionOption.stream, WildcardMentionOption.topic]),
+        ('topic',     channelNarrow, [WildcardMentionOption.topic]),
+        ('topic etc', topicNarrow,   <WildcardMentionOption>[]),
+
+        ('a',         dmNarrow,      [WildcardMentionOption.all]),
+        ('every',     dmNarrow,      [WildcardMentionOption.everyone]),
+        ('channel',   dmNarrow,      <WildcardMentionOption>[]),
+        ('stream',    dmNarrow,      <WildcardMentionOption>[]),
+        ('topic',     dmNarrow,      <WildcardMentionOption>[]),
+      ];
+
+      for (final (String query, Narrow narrow, List<WildcardMentionOption> wildcardOptions) in testCases) {
+        test('for "$query" query in ${narrow.runtimeType} narrow wildcards are $wildcardOptions', () async {
+          check(getWildcardOptionsFor(query, narrow: narrow)).deepEquals(wildcardOptions);
+        });
+      }
+
+      test('no wildcards for a silent mention', () {
+        check(getWildcardOptionsFor('', isSilent: true, narrow: channelNarrow)).isEmpty();
```

Comment:
nit: line too long: key information (the `.isEmpty`) past 80 columns:
```suggestion
        check(getWildcardOptionsFor('', isSilent: true, narrow: channelNarrow))
          .isEmpty();
```